### ISSUE TITLE
Re-formating Parsing

### ISSUE THREAD

**Issue by picatz** [picatz/command_lion#3]
Command Lion's parsing is fun and really easy to use; and, sadly easy to abuse:

This needs to be re-considered.

```ruby
CommandLion::App.run do
  name "Hello World"
  command :hello_world do
    flag "--hello-world"
    action do
      puts "Hello World!"
    end
    option :example do
      flag "--example"
      action do
        puts "Example worked!"
      end
    end
  end
end
```

## The Bug
It's simple in nature, but can cause some problems for people I'm sure.

The command works.
```shell
$ hello_world --hello-world
Hello World!
```

The option for the command works.
```shell
$ hello_world --hello-world --example
Hello World!
Example worked!
```

But, you can also call the option without having called the root command because the parsing is lazily done, essentially:
```shell
$ hello_world --example
Example worked!
```

### Lesson Learned
Probs don't make your own parser kids. 😹

**Comment by picatz** [comment]
mfw I'm just getting to looking at fixing this bug

**Comment by picatz** [comment]
Fixed in latests version *2.0.0* https://github.com/picatz/command_lion/releases/tag/v2.0.0
